configure

```diff
@@ -1062,6 +1067,28 @@ def configure_static(o):
         o['libraries'] += ['-static-libasan']
 
 
+def configure_gold(o):
+  try:
+    proc = subprocess.Popen(shlex.split('ld.gold') + ['-v'], stdin=subprocess.PIPE, 
+                            stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    match = re.match(r"(GNU gold) .* ([0-9]\.[0-9]+)",
+                         proc.communicate()[0])
+    if match:
+      gold_version = match.group(2)
+      if gold_version > '1.1':
+        o['variables']['gold_linker'] = 'true'
+        if options.section_reordering:
+          o['variables']['goldl_function_reorder'] = options.section_reordering
+        else:
+          o['variables']['goldl_function_reorder'] = os.path.realpath('tools/gold_linker_section_reordering.txt')
+        if(flavor in  ('linux', 'freebsd', 'openbsd')):
+          o['cflags'] += ["-fuse-ld=gold  -ffunction-sections"]
+          o['ldflags'] += ["-fuse-ld=gold -Wl,--section-ordering-file=" + o['variables']['goldl_function_reorder']] 
```

Comment:
Can you use single quotes?

I don't think this actually works right now.  You are adding single arguments to cflags and ldflags instead of two separate ones.  It should probably look like this:
```python
o['cflags'] += ['-fuse-ld=gold', '-ffunction-sections']
o['ldflags'] += ['-fuse-ld=gold', '-Wl,--section-ordering-file=...']
```
But that aside, hacking the flags straight into cflags and ldflags is somewhat horrible.  That should really be done in common.gypi:
```diff
diff --git a/common.gypi b/common.gypi
index d152c81498..bc73591391 100644
--- a/common.gypi
+++ b/common.gypi
@@ -42,6 +42,8 @@
     # Don't use ICU data file (icudtl.dat) from V8, we use our own.
     'icu_use_data_file_flag%': 0,
 
+    'gold_section_ordering_file%': '',  # export this from configure
+
     'conditions': [
       ['GENERATOR=="ninja"', {
         'OBJ_DIR': '<(PRODUCT_DIR)/obj',
@@ -431,7 +433,14 @@
         'ldflags': [
           '-Wl,--export-dynamic',
         ],
-      }]
+      }],
+      ['gold_section_ordering_file!=""', {
+        'cflags': [ '-fuse-ld=gold', '-ffunction-sections' ],
+        'ldflags': [
+          '-fuse-ld=gold',
+          '-Wl,--section-ordering-file=<(gold_section_ordering_file)',
+        ],
+      }],
     ],
   }
 }
```

Reply:
`-I../../common   -fuse-ld=gold  -ffunction-sections -pthread -Wall -Wextra -Wno-unused-parameter -m64 -fno-strict-aliasing -m64 -fdata-sections -O3 -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -MMD -MF  `

This is what it shows up as on the Make output. But I think separating it with commas would work better. 

I agree with, it's horrible to hack  straight into ldflags and cflags. 
I just couldn't figure out a way to export the value of `gold_section_ordering_file%`  into common.gypi which is a GYP include file.

I got an error saying symbol `gold_section_ordering_file` is not found.. 
I will try with the 
```.diff
+    'gold_section_ordering_file%': '',  # export this from configure
+
```